ql/src/java/org/apache/hadoop/hive/ql/parse/relnodegen/LateralViewPlan.java

```diff
@@ -256,9 +256,11 @@ private RelDataType getRetType(RelOptCluster cluster, RelNode inputRel,
     Preconditions.checkState(retType.isStruct());
 
     // Add the type names and values from the udtf into the lists that will make up the
-    // return type.
+    // return type. Names need to be unique so add the table prefix
     allDataTypes.addAll(Lists.transform(retType.getFieldList(), RelDataTypeField::getType));
-    allDataTypeNames.addAll(columnAliases);
+    for (String s : columnAliases) {
+      allDataTypeNames.add(lateralTableAlias + "." + s);
```

Comment:
We don't have to worry about special characters.  This step takes place only after parsing and it should guarantee a unique value.  

I suppose it may be possible to create an alias for the lateral view even with special characters where the names aren't unique.  But if we do?  There's no way to catch it here.